test/integration/framework/etcd.go

```diff
@@ -128,7 +129,15 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	stop := func() {
-		cancel()
+		// try to exit etcd gracefully
+		cmd.Process.Signal(syscall.SIGTERM)
+		go func() {
+			select {
+			case <-ctx.Done():
+			case <-time.After(5 * time.Second):
+				cancel()
+			}
+		}()
```

Comment:
@liggitt @pohly other source of noise in integration, the context send a SIGKILL and etcd dies automatically.
etcd support graceful shutdown with SIGTERM, if it doesn´t exist we kill it after 5 seconds

Reply:
this only is used if you run the test with `go test` as in `go test -timeout 100s -run ^TestLivezAndReadyz$ k8s.io/kubernetes/test/integration/controlplane -v -count 1` , the CI job uses an external etcd